eMPRess_GUI_version1.py

```diff
@@ -0,0 +1,354 @@
+        dup_entry.grid(row=0, column=1, sticky="w")
+        trans_entry.grid(row=1, column=1, sticky="w")
+        loss_entry.grid(row=2, column=1, sticky="w")
+
+    def plot_cost_landscape(self):
+        """Plots the cost landscape using matplotlib and embeds the graph in a tkinter window."""    
+        # creates a new tkinter window 
+        plt_window = tk.Toplevel(self.master)
+        plt_window.geometry("550x550")
+        plt_window.title("Matplotlib Graph DEMO")
+        # creates a new frame
+        plt_frame = tk.Frame(plt_window)
+        plt_frame.pack(fill=tk.BOTH, expand=1)
+        plt_frame.pack_propagate(False)
+        # use empress wrapper
+        recon_input = empress.read_input(self.file_path)
+        cost_regions = empress.compute_cost_regions(recon_input, 0.5, 10, 0.5, 10)  
+        #cost_regions.draw_to_file('./examples/cost_poly.png')  # draw and save to a file
+        fig = cost_regions.draw()  # draw to figure (creates matplotlib figure)
+        canvas = FigureCanvasTkAgg(fig, plt_frame)
+        canvas.draw()
+        canvas.get_tk_widget().pack(side=tk.TOP, fill=tk.BOTH, expand=True)
+        toolbar = NavigationToolbar2Tk(canvas, plt_frame)
+        toolbar.update()
+        canvas.get_tk_widget().pack(side=tk.TOP)
+        # updates the DTL costs using the x,y coordinates clicked by the user inside the graph 
+        # otherwise pops up a warning message window
+        fig.canvas.callbacks.connect('button_press_event', self.get_xy_coordinates)
+
+    def get_xy_coordinates(self, event):
+        """Updates the DTL costs when user clicks on the matplotlib graph, otherwise pops up a warning message window."""
+        if event.inaxes is not None:
+            self.dup_cost.set("1")
+            self.trans_cost.set(str(round(event.ydata, 2)))
+            self.loss_cost.set(str(round(event.xdata, 2)))
+            # enables the next step, viewing reconciliation result
+            self.has_dup_cost = True
+            self.has_trans_cost = True
+            self.has_loss_cost = True
+            self.enable_recon_btn()
+        else:
+            messagebox.showinfo("Warning", "Please click inside the axes bounds.")
+
+    def enable_recon_btn(self):
+        """Enables the next step, viewing reconciliation results, after DTL costs are set correctly."""
+        if self.has_dup_cost == True and self.has_trans_cost == True and self.has_loss_cost == True:
+            self.compute_recon_button.configure(state=tk.NORMAL)
+
+    def check_dup_input(self, *args):
+        """Checks and updates the duplication cost when user enters a value in the entry box."""
+        dupValue_without_decimal_point = self.dup_cost.get().replace(".", "")
+        if str.isdigit(dupValue_without_decimal_point) and ("-" not in dupValue_without_decimal_point):
+                self.dup_cost.set(str(round(float(self.dup_cost.get()), 2)))
+                self.has_dup_cost = True
```

Comment:
I think we could check whether we have duplication cost by checking whether
```python
self.dup_cost is not None
```

Reply:
I tried this. If I don't set any value for an empty entry box, its value will be "" instead of None. Then I went to initialize all entry boxes to be None, and I realized that each entry box will actually display "None" at the very beginning. What do you think? :))

Reply:
Oh, actually, this will still enable the next button if the "None" is not changed to a valid number. (I checked for self.xx_cost is not None for all three).